src/dex/fx-protocol-rusd/fx-protocol-rusd.ts

```diff
@@ -0,0 +1,314 @@
+              'fTokenRedeemFeeRatio',
+              [],
+            ),
+          },
+          {
+            target: this.config.weETHOracleAddress,
+            callData: this.weETHOracleIface.encodeFunctionData(
+              'latestAnswer',
+              [],
+            ),
+          },
+        ])
+        .call({}, blockNumber);
+
+    const redeemFee = BigInt(
+      this.rUSDMarketIface.decodeFunctionResult(
+        'fTokenRedeemFeeRatio',
+        data.returnData[0],
+      )[0],
+    ).toString();
+
+    const weETHPrice = BigInt(
+      this.weETHOracleIface.decodeFunctionResult(
+        'latestAnswer',
+        data.returnData[1],
+      )[0],
+    ).toString();
+
+    await Promise.all([
+      this.fxProtocolRusdPool.initialize(blockNumber, {
+        state: {
+          nav: '1000000000000000000',
+          redeemFee,
+          weETHPrice,
+        },
+      }),
+    ]);
+  }
+
+  getConfig() {
+    return this.config;
+  }
+
+  is_weETH(token: string) {
+    return token.toLowerCase() === this.config.weETHAddress.toLowerCase();
+  }
+
+  is_rUSD(token: string) {
+    return token.toLowerCase() === this.config.rUSDAddress.toLowerCase();
+  }
+
+  is_rUSD_swap_token(srcToken: string, destToken: string) {
+    return (
+      (this.is_weETH(srcToken) && this.is_rUSD(destToken)) ||
+      (this.is_rUSD(srcToken) && this.is_weETH(destToken))
+    );
+  }
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters() {
+    return null;
+  }
+
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    if (!this.is_rUSD_swap_token(srcToken.address, destToken.address)) {
+      return [];
+    }
+    return [`${this.dexKey}_${this.config.rUSDAddress}`];
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<FxProtocolData>> {
+    if (side == SwapSide.BUY) {
+      // note: BUY is not supported
+      return null;
+    }
+
+    const isRUSDSwapToken = this.is_rUSD_swap_token(
+      srcToken.address,
+      destToken.address,
+    );
+    if (!isRUSDSwapToken) {
+      return null;
+    }
+    const pool = this.fxProtocolRusdPool;
+    if (!pool.getState(blockNumber)) return null;
+
+    const is_redeem = this.is_weETH(destToken.address);
+    const unitIn = BI_POWS[18];
+    const unitOut = pool.getPrice(blockNumber, unitIn, is_redeem);
+    const amountsOut = amounts.map(_amountIn =>
+      pool.getPrice(blockNumber, _amountIn, is_redeem),
+    );
+
+    return [
+      {
+        unit: unitOut,
+        prices: amountsOut,
+        data: {},
+        poolAddresses: [this.config.rUSDAddress],
+        exchange: this.dexKey,
+        gasCost: 1000000,
```

Comment:
1 mil seems too high, lets look at examples and use the actual cost

Reply:
yes, its a actual gas cost, on average one swap take 960000-1115220 gas

Reply:
same for weETH<>rUSD and rUSD<>arUSD swaps